relays/bin-substrate/src/cli/detect_equivocations.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+use async_trait::async_trait;
+use relay_substrate_client::ChainWithTransactions;
+use structopt::StructOpt;
+use strum::{EnumString, EnumVariantNames, VariantNames};
+
+use crate::bridges::{
+	kusama_polkadot::{
+		kusama_headers_to_bridge_hub_polkadot::KusamaToBridgeHubPolkadotCliBridge,
+		polkadot_headers_to_bridge_hub_kusama::PolkadotToBridgeHubKusamaCliBridge,
+	},
+	rialto_millau::{
+		millau_headers_to_rialto::MillauToRialtoCliBridge,
+		rialto_headers_to_millau::RialtoToMillauCliBridge,
+	},
+	rialto_parachain_millau::millau_headers_to_rialto_parachain::MillauToRialtoParachainCliBridge,
+	rococo_wococo::{
+		rococo_headers_to_bridge_hub_wococo::RococoToBridgeHubWococoCliBridge,
+		wococo_headers_to_bridge_hub_rococo::WococoToBridgeHubRococoCliBridge,
+	},
+};
+use substrate_relay_helper::equivocation;
+
+use crate::cli::{bridge::*, chain_schema::*};
+
+/// Start equivocation detection loop.
+#[derive(StructOpt)]
+pub struct DetectEquivocations {
+	#[structopt(possible_values = DetectEquivocationsBridge::VARIANTS, case_insensitive = true)]
+	bridge: DetectEquivocationsBridge,
+	#[structopt(flatten)]
+	source: SourceConnectionParams,
+	#[structopt(flatten)]
+	source_sign: SourceSigningParams,
+	#[structopt(flatten)]
+	target: TargetConnectionParams,
+}
+
+#[derive(Debug, EnumString, EnumVariantNames)]
+#[strum(serialize_all = "kebab_case")]
+/// Equivocations detection bridge.
+pub enum DetectEquivocationsBridge {
+	MillauToRialto,
```

Comment:
https://github.com/paritytech/parity-bridges-common/commit/97133678203c1688d1aee6801de0c2788eecd691#r125925525

Reply:
(sorry - I've left an original comment for commit, not for the PR)

Reply:
Answered there.

Reply:
Fixed. Starting a prometheus endpoint for the ED now, with no metrics. So that we can see if the loop is up. Thanks for spotting this !